docker/src/main/java/clocker/docker/location/DockerContainerLocation.java

```diff
@@ -269,13 +273,100 @@ private void parseDockerCallback(String commandString) {
     }
 
     @Override
+    public int copyTo(final Map<String,?> props, final InputStream src, final long filesize, final String destination) {
+        Map<String,?> nonPortProps = Maps.filterKeys(props, Predicates.not(Predicates.containsPattern("port")));
+        boolean entitySsh = Boolean.TRUE.equals(entity.config().get(DockerContainer.DOCKER_USE_SSH));
+        boolean dockerSsh = Boolean.TRUE.equals(getOwner().config().get(DockerContainer.DOCKER_USE_SSH));
+        if (entitySsh && dockerSsh) {
+            return super.copyTo(nonPortProps, src, filesize, destination);
+        } else {
+            return copyTo(props, src, destination);
+        }
+    }
+
+    @Override
+    public int copyTo(final Map<String,?> props, final InputStream src, final String destination) {
+        Map<String,?> nonPortProps = Maps.filterKeys(props, Predicates.not(Predicates.containsPattern("port")));
+        boolean entitySsh = Boolean.TRUE.equals(entity.config().get(DockerContainer.DOCKER_USE_SSH));
+        boolean dockerSsh = Boolean.TRUE.equals(getOwner().config().get(DockerContainer.DOCKER_USE_SSH));
+        if (entitySsh && dockerSsh) {
+            return super.copyTo(nonPortProps, src, destination);
+        } else {
+            File tmp = null;
+            try {
+                tmp = File.createTempFile(dockerContainer.getId(), Urls.getBasename(destination));
+                hostMachine.copyTo(nonPortProps, src, tmp.getAbsolutePath());
```

Comment:
No, the `File` object is just a representation. We are then copying the source to that temp file on the Docker host like you suggest. The usage is a bit misleading, I agree. I got the implementation wrong myself the first time, because of that 😸